clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3777,4 +3788,25 @@ protected boolean shouldUpdateUpstreamOffset(PubSubMessage<KafkaKey, KafkaMessag
   protected void maybeSendIngestionHeartbeat() {
     // No op, heartbeat is only useful for L/F hybrid stores.
   }
+
+  /**
+   * This function is checking the following conditions:
+   * 1. Whether the version topic exists or not.
+   * 2. Whether the real-time topic in local colo exists or not for hybrid store.
+   */
+  public boolean isProducingTopicHealthy() {
+    if (isDaVinciClient) {
+      /**
+       * DaVinci doesn't produce to any topics.
+       */
+      return true;
+    }
+    if (!topicManagerRepository.getTopicManager().containsTopic(this.versionTopic)) {
```

Comment:
Is it better to have a timestamped based retries here? like if we cannot find the topic for over some period of checks then only declare its unhealthy? nit: also can we rename to `isProducingTopicExists`

Reply:
Yeah, I replied with the similar idea to @lluwm 's comment